### ISSUE TITLE
Add a service backend to store passkeys in TPM (or other hardware) like Windows Hello.

### ISSUE THREAD

**Issue by IPlayZed** [systemd/systemd#30402]
### Component

homectl, pam_systemd, pam_systemd_home, systemd, systemd-homed

### Is your feature request related to a problem? Please describe

It would nice if there was a unified backend for desktop environments to implement a passkey storage into keyrings/wallets to store passkeys to hardware backed storage like TPM or something similar. This would greatly help adoption as an overwhelming amount of Linux systems use SystemD as their base, compared to seperate implementations.

### Describe the solution you'd like

A service with the corresponding *ctl terminal interface is created to easily store passkeys. Desktop environments should be able to access this service via a secure channel, like D-Bus, to offer some GUI management. Applications like web browsers should also be able to implement and use this interface on Linux to easily create and access these passkeys.
Passkeys should be configurable to be tied to a given TPM policy through pcrlock added in v255.
I would also consider tying these passkeys in some form to human-user accounts specifically, if possible as an option.

### Describe alternatives you've considered

I am using the Bitwarden browser extension, but that is not hardware backed, so the use case is different. I am also using USB security keys or my phone in addition as passkeys, but there is no similar solution like Windows Hello, which ties the passkey to a measured environment specific to my Linux installation defined by my TPM PCR policy.
Creating a separate project for this or solution specific to some application would hurt adoption, as it would not be flexible.

### The systemd version you checked that didn't have the feature you are asking for

255

**Comment by IPlayZed** [comment]
I was thinking about if systemd-credentials can be expanded or used for this scenario?

**Comment by abhi-deshpande** [comment]
Yes. This would be a nice thing to have and would eliminate the need to store passkeys in password managers.

**Comment by IPlayZed** [comment]
Thank you for supporting the idea, unfortunately implementing this is way over my head, but hopefully someone will see this with the right knowledge and start an actual PR for it. 😄
